src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,244 @@
+        {
+            if (calleeMethodDeclarationSyntaxNode is MethodDeclarationSyntax declarationSyntax)
+            {
+                var blockSyntaxNode = declarationSyntax.Body;
+                // 1. If it is an ordinary method with block
+                if (blockSyntaxNode != null)
+                {
+                    var blockStatements = blockSyntaxNode.Statements;
+                    return blockStatements.Count == 1 && ShouldStatementBeInlined(blockStatements[0]);
+                }
+                else
+                {
+                    // 2. If it is an Arrow Expression
+                    var arrowExpressionNodes = declarationSyntax.ExpressionBody;
+                    return arrowExpressionNodes != null;
+                }
+            }
+
+            return false;
+        }
+
+        protected override IParameterSymbol? GetParameterSymbol(SemanticModel semanticModel,ArgumentSyntax argumentSyntaxNode, CancellationToken cancellationToken)
+            => argumentSyntaxNode.DetermineParameter(semanticModel, allowParams: true, cancellationToken);
+
+        protected override SyntaxNode? GetInlineStatement(SyntaxNode calleeMethodDeclarationSyntaxNode)
+        {
+            var declarationSyntax = (MethodDeclarationSyntax)calleeMethodDeclarationSyntaxNode;
+            SyntaxNode? inlineSyntaxNode = null;
+            var blockSyntaxNode = declarationSyntax.Body;
+            // 1. If it is a ordinary method with block
+            if (blockSyntaxNode != null)
+            {
+                var blockStatements = blockSyntaxNode.Statements;
+                if (blockStatements.Count == 1)
+                {
+                    inlineSyntaxNode = GetExpressionFromStatementSyntaxNode(blockStatements[0]);
+                }
+
+                return inlineSyntaxNode;
+            }
+            else
+            {
+                // 2. If it is using Arrow Expression
+                var arrowExpressionNode = declarationSyntax.ExpressionBody;
+                if (arrowExpressionNode != null)
+                {
+                    inlineSyntaxNode = arrowExpressionNode.Expression;
+                }
+
+                return inlineSyntaxNode;
+            }
+
+            // A check has been done before to make sure there is one arrow expression or block statement.
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        private static SyntaxNode? GetExpressionFromStatementSyntaxNode(StatementSyntax statementSyntax)
+            => statementSyntax switch
+            {
+                ReturnStatementSyntax returnStatementSyntax => returnStatementSyntax.Expression,
+                ExpressionStatementSyntax expressionStatementSyntax => expressionStatementSyntax.Expression,
+                _ => null
+            };
+
+        protected override bool ShouldCheckTheExpressionPrecedenceInCallee(SyntaxNode syntaxNode)
+            => s_syntaxKindsNeedsToCheckThePrecedence.Any(syntaxNode.IsKind);
```

Comment:
```suggestion
            => s_syntaxKindsNeedsToCheckThePrecedence.Contains(syntaxNode.Kind());
```

No allocations with that fomr.

Also, consider a set instead of an array here.